tinylfu.go

```diff
@@ -5,55 +5,119 @@
 package tinylfu
 
 import (
-	"container/list"
 	"github.com/dgryski/go-metro"
 )
 
-type T struct {
-	c       *cm4
-	bouncer *doorkeeper
-	w       int
-	samples int
-	lru     *lruCache
-	slru    *slruCache
-	data    map[string]*list.Element
+type T[V any] struct {
+	c              *cm4
+	bouncer        *doorkeeper
+	w              int
+	samples        int
+	lru            *lruCache[V]
+	slru           *slruCache[V]
+	data           map[string]*Element[slruItem[V]]
+	hits           uint64
+	misses         uint64
+	lruPct         float32
+	interval       int
+	step           int
+	percentage     float32
+	wentUp         bool
+	madeStepBigger bool
+	lastSuccess    float32
+	size           int
+	withResize     bool
 }
 
-func New(size int, samples int) *T {
+func New[V any](size int, samples int, withResize bool) *T[V] {
```

Comment:
Rather than changing the API for `New` here, maybe have `New` and `NewWithResize` ? (Both can call into an unexported `newCache` constructor that sets the flags as appropriate.